src/analysis/retail/evoker/preservation/modules/core/MasteryEffectiveness.tsx

```diff
@@ -54,14 +57,16 @@ class MasteryEffectiveness extends Analyzer {
 
   private prevokerHealthPercent: number = 0;
   private prevokerMaxHealth: number = 0;
-  private anomalyShieldApplications: Map<number, ShieldInfo | null> = new Map();
-  private twinGuardianShieldApplications: Map<number, ShieldInfo | null> = new Map();
+  private anomalyShieldApplications: Map<[number, number], ShieldInfo | null> = new Map();
+  private twinGuardianShieldApplications: Map<[number, number], ShieldInfo | null> = new Map();
```

Comment:
this isn't going to work the way you want. `Map` keys are compared using `Object.is`, which is effectively reference equality. Compare the output of `Object.is(1, 1)`, `Object.is([1, 1], [1, 1])`, and `const x = [1, 1]; Object.is(x, x)`. Only the 1st and 3rd should be true. The 2nd should be false, because each time you create the array it has a different reference---even if it has identical contents.

Instead, you should use a `string` key for this, and use a function to generate the keys like:

```ts
// FIXME: not sure what the params are off-hand
function shieldKey(a: number, b: number): string {
  return `${a}-${b}`;
}
```

Reply:
Reading through this, I'm actually not sure that you need the 2nd part of the key. Looks like the key is `[targetID, spellID]` but `spellID` is effectively constant---it is always the same for TA, and always the same for TG, so you can just omit (like the old code did)

Reply:
You're correct, misread my first time looking at this and had it in my head both shields were being pushed to the same map. Fixed